submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/model/FilesSource.kt

```diff
@@ -15,8 +14,26 @@ interface FilesSource {
     fun readText(filePath: String): String
 }
 
-class ListFilesSource(private val files: List<ResourceFile>) : FilesSource {
+class MixedFilesSource(
+    private val listFiles: ListFilesSource,
+    private val pathFiles: PathFilesSource
+) : FilesSource {
+    override fun exists(filePath: String) = listFiles.exists(filePath).or(pathFiles.exists(filePath))
+
+    override fun getInputStream(filePath: String) =
+        executeBySource(filePath, { listFiles.getInputStream(filePath) }, { pathFiles.getInputStream(filePath) })
```

Comment:
Consider the following aproach

`= exists(filePath).fold({ listFiles.getInputStream(filePath) }, {pathFiles.getInputStream(filePath) } }})`